meshroom/nodes/aliceVision/SketchfabUpload.py

```diff
@@ -0,0 +1,200 @@
+            description='''verbosity level (critical, error, warning, info, debug).''',
+            value='info',
+            values=['critical', 'error', 'warning', 'info', 'debug'],
+            exclusive=True,
+            uid=[],
+        ),
+    ]
+
+    def resolvedPaths(self, inputFiles):
+        paths = []
+        for inputFile in inputFiles:
+            for f in glob.glob(inputFile.value):
+                paths.append(f)
+        return paths
+    
+    def upload(self, apiToken, modelFile, data, chunk):
+        modelEndpoint = 'https://api.sketchfab.com/v3/models'
+        f = open(modelFile, 'rb')
+        file = {'modelFile': (os.path.basename(modelFile), f.read()), **data}
+        (files, contentType) = requests.packages.urllib3.filepost.encode_multipart_formdata(file)
+        headers = {'Authorization': 'Token {}'.format(apiToken), 'Content-Type': contentType}
+        body = BufferReader(files, progressUpdate, cb_kwargs={'logManager': chunk.logManager})
+        chunk.logger.info('Uploading...')
+        try:
+            r = requests.post(
+                modelEndpoint, **{'data': body, 'headers': headers})
+            f.close()
+            chunk.logManager.completeProgressBar()
+        except requests.exceptions.RequestException as e:
+            f.close()
+            chunk.logger.error(u'An error occured: {}'.format(e))
+            raise RuntimeError() 
+        if r.status_code != requests.codes.created:
+            chunk.logger.error(u'Upload failed with error: {}'.format(r.json()))
+            raise RuntimeError()
+
+    def processChunk(self, chunk):
+        chunk.logManager.waitUntilCleared()
+        chunk.logger.setLevel(chunk.logManager.textToLevel(chunk.node.verboseLevel.value))
+        
+        if not chunk.node.inputFiles:
+            chunk.logger.warning('Nothing to upload')
+            return
+        if len(chunk.node.title.value) > 48:
+            chunk.logger.error('Title cannot be longer than 48 characters.')
+            raise RuntimeError()
+        if len(chunk.node.description.value) > 1024:
+            chunk.logger.error('Description cannot be longer than 1024 characters.')
+            raise RuntimeError()
+        if chunk.node.apiToken.value == '':
+            chunk.logger.error('Need API token.')
+            raise RuntimeError()
+        data = {
+            'name': chunk.node.title.value,
+            'description': chunk.node.description.value,
+            'license': chunk.node.license.value,
+            'isPublished': chunk.node.isPublished.value,
+            'isInspectable': chunk.node.isInspectable.value
+        }
+        # pack files into .zip to reduce file size and simplify process
+        uploadFile = os.path.join(chunk.node.internalFolder, 'temp.zip')
+        files = self.resolvedPaths(chunk.node.inputFiles.value)
+        zf = zipfile.ZipFile(uploadFile, 'w')
+        for file in files:
+            zf.write(file, os.path.basename(file))
```

Comment:
Does this work when we connect Texture.outputTextures which contains the simplified regex: "texture_*.png"?

Reply:
Yes because it goes through `resolvedPaths` which uses glob, like in the publish node. However, it does not work when Texture.output is connected because only the folder is copied, not the files in the folder. I will fix this